src/network/ifaddrs_patch.h

```diff
@@ -0,0 +1,34 @@
+/*!
+* Copyright (c) 2009, Jay Loden, Dave Daeschler, Giampaolo Rodola.
+ * Licensed under the BSD 3-Clause License.
+ * See https://github.com/giampaolo/psutil/blob/master/LICENSE
+ */
+
+/*
+- https://lists.samba.org/archive/samba-technical/2009-February/063079.html
+- https://github.com/giampaolo/psutil/blob/master/psutil/arch/solaris/v10/ifaddrs.h
+*/
+
+#ifndef __IFADDRS_H__
+#define __IFADDRS_H__
```

Comment:
For the consistency with the rest cpp codebase.
```suggestion
#ifndef LIGHTGBM_NETWORK_IFADDRS_PATCH_H_
#define LIGHTGBM_NETWORK_IFADDRS_PATCH_H_
```

Reply:
ok yeah you're right. The `__IFADDRS_H__` was just directly copied from the `psutil` patch.